routes/offerings.js

```diff
@@ -0,0 +1,126 @@
+const express = require("express");
+const router = express.Router();
+const Offering = require("../models/Offering");
+const Posting = require("../models/Posting");
+const Notifications = require("../utils/notificationsUtil");
+
+// http://localhost/api/offerings/{offeringId}/accept ///////////////////////////////////////////////////////
+// POST
+// Accepts the indicated offering
+router.post("/:offeringId/accept", async (req, res) => {
+  try {
+    const offering = await Offering.findOneAndUpdate(
+      {
+        _id: req.params.offeringId,
+      },
+      {
+        $set: { status: "ACCEPTED" },
+      },
+      {
+        new: true,
+      }
+    );
+
+    await Notifications(
+      offering.userId,
+      "offeringAccepted",
+      "Your offering has been accepted!"
+    );
+
+    const allPostingOfferings = await Posting.findOneAndUpdate(
+      {
+        _id: offering.postingId,
+      },
+      {
+        $set: { active: false },
+      }
+    ).offerings;
+
+    allPostingOfferings.forEach(async (offeringId) => {
+      if (offeringId !== req.params.offeringId) {
```

Comment:
This check doesn't seem to be working. When I check the offerings for the posting, all of them (including the one I accepted) is set to `REJECTED`. Logging the `offeringId` inside the if statement, I still get the `req.params.offeringId` printed.